ThreeBotPackages/tfgrid_solutions/tfgrid_solutions/chatflows/monitoring_solution_deploy.py

```diff
@@ -0,0 +1,340 @@
+        self.user_form_data["Solution name"] = j.sal.reservation_chatflow.solution_name_add(self, self.model)
+
+    @j.baseclasses.chatflow_step(title="Access keys")
+    def public_key_get(self):
+        self.user_form_data["Public key"] = self.upload_file(
+            """Please add your public ssh key, this will allow you to access the deployed containers using ssh.
+                    Just upload the file with the key""",
+            required=True,
+        ).split("\n")[0]
+
+    @j.baseclasses.chatflow_step(title="Prometheus container resources")
+    def prometheus_container_resources(self):
+        form = self.new_form()
+        cpu = form.int_ask(
+            "Please add how many CPU cores are needed for the Prometheus container", default=1, required=True
+        )
+        memory = form.int_ask("Please add the amount of memory in MB", default=1024, required=True)
+        rootfs_type = form.single_choice(
+            "Select the storage type for your root filesystem", ["SSD", "HDD"], default="SSD"
+        )
+        rootfs_size = form.int_ask("Choose the amount of storage for your root filesystem in MiB", default=256)
+        form.ask()
+        self.user_form_data["Prometheus CPU"] = cpu.value
+        self.user_form_data["Prometheus Memory"] = memory.value
+        self.user_form_data["Prometheus Root filesystem Type"] = str(rootfs_type.value)
+        self.user_form_data["Prometheus Root filesystem Size"] = rootfs_size.value
+
+        self.prometheus_query["mru"] = math.ceil(self.user_form_data["Prometheus Memory"] / 1024)
+        self.prometheus_query["cru"] = self.user_form_data["Prometheus CPU"]
+        storage_units = math.ceil(self.user_form_data["Prometheus Root filesystem Size"] / 1024)
+        if rootfs_type.value == "SSD":
+            self.prometheus_query["sru"] = storage_units
+        else:
+            self.prometheus_query["hru"] = storage_units
+
+    @j.baseclasses.chatflow_step(title="Prometheus volume details")
+    def prometheus_volume_details(self):
+        form = self.new_form()
+        vol_disk_type = form.drop_down_choice(
+            "Please choose the type of disk for the volume to be attached to the Prometheus container",
+            ["SSD", "HDD"],
+            required=True,
+            default="SSD",
+        )
+        vol_disk_size = form.int_ask("Please specify the volume size", required=True, default=10)
+        form.ask()
+        self.user_form_data["Prometheus Volume Disk type"] = vol_disk_type.value
+        self.user_form_data["Prometheus Volume Size"] = vol_disk_size.value
+
+    @j.baseclasses.chatflow_step(title="Grafana container resources")
+    def grafana_container_resources(self):
+        form = self.new_form()
+        cpu = form.int_ask(
+            "Please add how many CPU cores are needed for the Grafana container", default=1, required=True
+        )
+        memory = form.int_ask("Please add the amount of memory in MB", default=1024, required=True)
```

Comment:
Are the proposed default value for CPU and memory viable for a grafana container ?

Reply:
Yes and verified with @muhamadazmy as well.

Reply:
Ok scratch there, seems 1GiB is more then enough.

from https://grafana.com/docs/grafana/latest/installation/requirements/
> Grafana does not use a lot of resources and is very lightweight in use of memory and CPU.
Minimum recommended memory: 255 MB Minimum recommended CPU: 1